DD-28 API system information

armshare opened the pull request.

Commit:
Featue Issue#3

add Enhance create project error message

Commit:
feature Issue#3

add Enhance create project error message

Commit:
Merge branch 'feature/Issue#3' into develop

Commit:
fix issue#3 from retrive DB to File system

Commit:
fix issue#3

change retrieve DB to retrieve FileSystem

Commit:
reformat

Commit:
fix follow recommend

Commit:
fix follow recommend

Commit:
DD-3 Enhance create project error message

Update appseting   AppController

Commit:
..

Commit:
.

Commit:
DD-3 Update AppControllerBase

Commit:
DD3- Add AppController

Commit:
DD3-Change Error Message Template Not found

Commit:
DD3-Change error message return Template Notfound

Commit:
DD3 - Change git .ignore update AppController

Add .ignore *.csproj.user
Update variable ObjectName;
Remove Method BadRequest

Commit:
DD3-Remove file .User

Commit:
DD3 - Merge

Commit:
Merge branch 'develop' into feature/Issue#3

Commit:
merge remote

Commit:
update appController

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

Commit:
Merge branch 'develop' into feature/Issue#3

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into feature/Issue#3

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

surrealist committed:
DD-30 Bump version GH Action (#31)

surrealist committed:
initial versions

surrealist committed:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

surrealist committed:
Bump version - Add PAT

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

actions-user committed:
Bump version

surrealist committed:
Bump version - stop the loop

surrealist committed:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

actions-user committed:
Bump version

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

surrealist committed:
hotfix - EF Proxies

actions-user committed:
Bump version

surrealist committed:
Bump version - halt

surrealist committed:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

actions-user committed:
Bump version

S43493 committed:
DD-10 Update project info

* Issue#10

Update project info

* Edit Code from comment

* .

* DD-10 Edit Code

* DD-10 Edit ProjectResponse

* EDIT CODE

* EF Proxies" Version="6.0.18"

---------

Co-authored-by: NUTTANUN PUMSA-ARD <nuttanunp@scbam.onmicrosoft.com>

actions-user committed:
Bump version

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

Commit:
Develop - Merge from upstream

armshare committed:
DD-28 API system information

add SystemController
add Model Respones

armshare committed:
Merge branch 'develop' into feature/DD-28

armshare committed:
DD-28

fix config Appsetting.Dev
add Datetime utc + localTimeZone
SystemController

Commit:
Merge from Remote

Commit:
Merge branch 'develop' into feature/DD-28

Commit:
DD-28

add get environment

Commit:
delete feature Issue#3

Commit:
Merge branch 'develop' into feature/DD-28

surrealist commented on DevDynamo.Web/Areas/ApiV1/Controllers/SystemController.cs:
```diff
@@ -0,0 +1,34 @@
+        {
+         
+            var v = GetType().Assembly.GetName().Version.ToString();
+            var envi = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
```
We should use the `EnvironmentName` of the `IWebHostEnvironment` injected into the controller. 

Access to the environment variable might not be correct in every case.

surrealist commented on DevDynamo.Web/Areas/ApiV1/Controllers/SystemController.cs:
```diff
@@ -0,0 +1,34 @@
+         
+            var v = GetType().Assembly.GetName().Version.ToString();
+            var envi = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+            DateTime utcTime = DateTime.UtcNow;
+            string localTime = utcTime.ToLocalTime().ToString("zzz");
+            string now = utcTime.ToString($"yyyy-MM-dd HH:mm:ss{localTime}");
```
combine UTC date/time with the timezone of the local. I don't think it's correct. 🤔

armshare replied:
ผมลองใช้ ทั้ง 
```
string utcTime = DateTime.UtcNow.ToString($"yyyy-MM-dd HH:mmzzz"); //2023-07-06 15:32+00:00
string localTime = DateTime.Now.ToString($"yyyy-MM-dd HH:mmzzz"); //2023-07-06 22:32+07:00
```
หรือมันมีมากกว่านี้อ่ะครับ

surrealist commented on DevDynamo.Web/Areas/ApiV1/Models/SystemResponse.cs:
```diff
@@ -0,0 +1,13 @@
+{
+  public class SystemResponse
+  {
+    public string? version { get; set; }
+    public string? environment { get; set; }
+    public string? now { get; set; }
```
We should name properties with CamelCase as it is a C# code. When it has returned from the API, it'll be converted to camelCase automatically.

surrealist commented:
Why this PR has so many commits? You might need an updated upstream version.

Commit:
Merge branch 'develop' of https://github.com/surrealist/dev-dynamo into develop

Commit:
Merge branch 'develop' into feature/DD-28

Commit:
DD-28 API system information

fix issue Get environment varible
Get stuck in in part Datetime format and get version display

Commit:
DD-28 API system information

reformat warning of "Dereference of a possibly null reference."

surrealist commented:
Thanks for the update. Let's talk tmr.

Commit:
DD-28 system information

update from code review
